src/handlers/psp.ts

```diff
@@ -349,8 +267,40 @@ const submitTransactions = async (state: State) => {
     []
   );
 
-  // TODO: start new lambdas for each providerSubscriptions array element
-  await processSubscriptions(providerSponsorSubscriptions, baseLogOptions);
+  const providerSponsorPromises = providerSponsorSubscriptionsArray.map(async (providerSponsorSubscriptions) =>
+    spawn({
+      providerSponsorSubscriptions,
+      baseLogOptions: baseLogOptions,
+      type: config.nodeSettings.cloudProvider.type,
+      stage: config.nodeSettings.stage,
+    })
+  );
+
+  const providerSponsorResults = await Promise.allSettled(providerSponsorPromises);
+
+  const groupedResults = providerSponsorResults.reduce(
+    (
+      acc: {
+        fulfilled: PromiseFulfilledResult<string>[];
+        rejected: PromiseRejectedResult[];
+      },
+      result
+    ) => {
+      if (result.status === 'fulfilled') {
+        acc.fulfilled.push(result);
+      }
+      if (result.status === 'rejected') {
+        acc.rejected.push(result);
+      }
+      return acc;
+    },
+    { fulfilled: [], rejected: [] }
+  );
+
+  utils.logger.info(
+    `Processed subscriptions; success: ${groupedResults.fulfilled.length}, failure: ${groupedResults.rejected.length}`,
+    baseLogOptions
+  );
```

Comment:
I don't think this is giving that much info tbh. It's a nice statistic but I would much rather know what subscriptions failed and why.

I think you should log the messages you're currently returning from lambda but not doing anything with. But I guess that can be done in the lambda and not here.

Reply:
> I think all subscription specific logs are handled in process-subscriptions.ts [here](https://github.com/api3dao/airkeeper/pull/53/files#diff-62a0c95e58afb4957b9bf3376a5ccf9c38745a1946518a36d00fb0bfd0fe06b2R62-R65), right?

Yes

> How about doing it like this to log possibly rejected promises (i.e. errors from the lambda)?

But wouldn't be better to log it in the lambdas instead of returning it and logging it here?

Reply:
> But wouldn't be better to log it in the lambdas instead of returning it and logging it here?

Yes for the successful cases, but I think we would still need to log (unhandled) invoke lambda errors from the spawn function in them main lambda: 

```ts
lambda.invoke(options, (err, data) => {
      // Reject invoke and (unhandled) handler errors
      if (err || data.FunctionError) {
        reject(err || data.FunctionError);
        return;
      }
...
```

Or did you have something else in mind?

Reply:
Yes, but this is different from what you wrote above. I think we should log success and expected failures in the lambda but of course, in case of the lambda invocation failure, we have to log that here.

Reply:
> Yes, but this is different from what you wrote above. I think we should log success and expected failures in the lambda but of course, in case of the lambda invocation failure, we have to log that here.

Just to make sure one more time :smile:, these are the changes:
1. we remove the `result.status === 'fulfilled'` block from the main lambda and leave the `rejected` block.
2. we log `Processing subscriptions for sponsorAddress.... finished` in the `process-subscriptions` lambda (and remove the `message` and `statusCode` fields as those are not needed anymore).

Reply:
Yes, let's do it that way :slightly_smiling_face: